src/components/app/Root.js

```diff
@@ -129,6 +143,12 @@ class ProfileViewWhenReadyImpl extends PureComponent<ProfileViewProps> {
   componentDidUpdate(prevProps) {
     if (prevProps.dataSource === 'none' && this.props.dataSource !== 'none') {
       this._retrieveProfileFromDataSource();
+    } else if (
+      this.props.dataSource === 'compare' &&
+      !prevProps.profilesToCompare &&
+      this.props.profilesToCompare
+    ) {
+      this.props.retrieveProfilesToCompare(this.props.profilesToCompare);
```

Comment:
note: This is a change that is included in a commit you already reviewed, that I had to do as part of the rebase